internal/handlers/pg/pgdb/query.go

```diff
@@ -247,20 +247,45 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
 	var p Placeholder
 
 	for k, v := range sqlFilters.Map() {
-		if len(k) != 0 && (k[0] == '$' || types.NewPathFromString(k).Len() > 1) {
+		if len(k) != 0 && k[0] == '$' {
 			// TODO $eq and $ne https://github.com/FerretDB/FerretDB/issues/1840
-			// TODO dot notation https://github.com/FerretDB/FerretDB/issues/1841
+			// TODO $gt and $lt https://github.com/FerretDB/FerretDB/issues/1875
 			continue
 		}
 
-		// don't iterate through array for _id keys to simplify the query
-		if k == "_id" {
+		operator := "->" // operator is the operator that is used to access the field. (->/#>)
+
+		var key any = k   // key can be either a string '"v"' or path '{v,foo}'
+		var prefix string // prefix is the first key in path, if the filter key is not a path - the prefix is empty
+
+		// If the key is in dot notation use path operator (#>)
+		if len(k) != 0 {
```

Comment:
>  Is this because we can have empty key like {"","foo"}?
Exactly, it's possible to have an empty key.

It's also worth to mention that if we try to put an empty key `""` inside the `NewPathFromString` function, it will panic, so we need to nest it inside the if statement

Reply:
>It's also worth to mention that if we try to put an empty key "" inside the NewPathFromString function, it will panic, so we need to nest it inside the if statement

Please create an issue for it or return the same error as compat :pray: Panic on something user can accidentally do is not ideal.

Reply:
I meant that this panic is expected as we don't want to use `NewPathFromString` function on empty strings. The goal of this if statement is to don't call the function to don't panic on empty keys. Sorry if I didn't clarify correctly

Reply:
I understand what you are trying to do here :hugs: 

But I'm concerned that `NewPathFromString` is panicing when it shouldn't. It is used in 20+ places and each time if we try to use it, we have to check for empty keys or it panics. It seems to me that `NewPathFromString` should return `BadValue` error. So I think you discovered a bug here. We should create an issue to handle it. Because other 20+ places don't have `len(k) != 0` logic to avoid panic. :see_no_evil:

Reply:
I see your point, and it seems that making empty check inside the path might be a good approach. 

I'm not sure about the "bug" part though. At least 9 of those calls are inside the tests, and it seems that we always try to check how operators and commands work with empty keys in test cases. 

>  don't have len(k) != 0 logic to avoid panic.

I think in most cases those check are made inside other functions (for example distinct command does it while fetching the parameter from document.

Reply:
I think this is related https://github.com/FerretDB/FerretDB/issues/1744 :+1: